### PR TITLE
chore: Use SemanticVersion instead of SoftwareVersion

### LINKED ISSUE #13432: Switch to using the PBJ version for all comparisons
Part of #13344
Since `SoftwareVersion` will no longer be part of an event, we cannot rely on it for comparisons. We should use the PBJ semantic version for all comparisons related to events.

### PR DESCRIPTION
**Pull request by timo0**
Fix #13432

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java [hunk 1]
```diff
@@ -227,8 +227,12 @@ public Instant getTimeCreated() {
 
     @Nullable
     @Override
-    public SoftwareVersion getSoftwareVersion() {
-        return hashedData.getSoftwareVersion();
+    public SemanticVersion getSoftwareVersion() {
+        if (hashedData.getSoftwareVersion() != null) {
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java:231]: i dont think the version should ever be null

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/Event.java [hunk 2]
```diff
@@ -74,5 +74,5 @@ default void forEachTransaction(final Consumer<Transaction> consumer) {
      * @return the software version
      */
     @Nullable
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/Event.java:76]: should be non null

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java [hunk 3]
```diff
@@ -272,7 +272,7 @@ public long getBirthRound() {
      */
     @Override
     @Nullable
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:274]: NonNull

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/TestingEventBuilder.java [hunk 4]
```diff
@@ -203,8 +204,8 @@ public TestingEventBuilder(@NonNull final Random random) {
      * @param softwareVersion the software version
      * @return this instance
      */
-    public @NonNull TestingEventBuilder setSoftwareVersion(@Nullable final SoftwareVersion softwareVersion) {
-        this.softwareVersion = softwareVersion;
+    public @NonNull TestingEventBuilder setSoftwareVersion(@Nullable final SemanticVersion softwareVersion) {
+        this.softwareVersion = new BasicSoftwareVersion(softwareVersion.major());
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/TestingEventBuilder.java:208]: Maybe for this use case we should introduce a new type that simply wraps the existing PBJ version

Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/TestingEventBuilder.java:208]: talked about

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/Event.java (hunk 2 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/Event.java]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java (hunk 1 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java (hunk 3 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:274]: done

Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:274]: not done

**Review by lpetrovic05 — DISMISSED** [review]

**Review by JeffreyDallas — DISMISSED** [review]

**Review by thomas-swirlds-labs — DISMISSED** [review]
LGTM

**Review by alittley — APPROVED** [review]

**Review by thomas-swirlds-labs — DISMISSED** [review]
LGTM. Thanks! @timo0 🎉

**Review by JeffreyDallas — DISMISSED** [review]

**Review by thomas-swirlds-labs — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @timo0 !

**Review by rbarkerSL — APPROVED** [review]
LGTM

**Review by mxtartaglia-sl — APPROVED** [review]

**Pull request merged**